draft/XXXX-multipart-upload/README.md

```diff
@@ -0,0 +1,115 @@
+---
+RFC: XXXX
+Author: Roni Choudhury
+Status: draft
+Created: 2020-08-04
+Last-Modified: 2020-08-04
+---
+
+# RFC XXXX: Multipart Table Upload
+
+This RFC describes how large table data can be safely and efficiently uploaded
+to Multinet, using a variation of the common multipart upload strategy common to
+many web server technologies.
+
+## Motivation
+
+Multipart upload is needed for the following reasons:
+
+1. As a practical matter, our hosting platform Heroku does not allow for
+   requests that take longer than thirty seconds to complete. Some moderately
+   sized CSV files already break this threshold, and so called "big data" tables
+   most certainly will. In order to upload such data files, we will need a way
+   to upload them in chunks.
+2. A long-running one-shot upload does not give the end user much feedback on
+   what is going on. Multipart upload would allow us to provide a more detailed
+   progress meter that can improve the user experience.
+3. If a long-running upload is interrupted for any reason, a multipart upload
+   strategy allows for resuming such uploads without losing the time already
+   spent in uploading one more initial chunks of the file.
+
+## Proposals
+
+### Summary of Approach
+
+The basic idea is as follows: the client initiates a new upload session for a
+given file; this action causes the server to create a new collection in the
+database. Then the client issues one or more "chunk" uploads, each sending a
```

Comment:
Which database would these new per-upload collections be stored? The `_system` database? Or maybe we want a separate `_upload` database? I also think we'll need a separate collection for tracking the status of each upload (one-to-one with the collections in whatever database upload collections are stored). Each document in this collection would track the metadata, status, etc. of each upload, and would point to a collection in the "uploads" database.

Reply:
I think an `_uploads` database is a good idea. Did we already figure out that collections can't contain metadata in ArangoDB? We might wind up with an `upload_mapping` collection in `_system` in that case (I think this is what you're describing).

Reply:
I don't believe that you can set arbitrary data on collections. What you're describing in the end of your comment is what I had in mind, although I'd prefer to get away from the semantics of "mapping" and just call it "uploads" or something, which also happens to reference a collection.

Reply:
Yeah I agree. You could even file an issue to refactor `workspace_mapping` to `workspaces` or something (though that would be pretty low priority).

Reply:
@waxlamp If we've reached a consensus here (which I think we have), can you add the points from this discussion into the RFC?

Reply:
Yes, I've been meaning to complete the work on this RFC. I'll try to focus on it today.

I'll also add the PR Mike's working on to the "reference implementation" section.

Reply:
It seems that the underscore in `_uploads` makes it an invalid database name -
https://www.arangodb.com/docs/stable/data-modeling-naming-conventions-database-names.html

Should we just go with `uploads` for the DB name then?

Reply:
That should be fine.